src/cache.js

```diff
@@ -140,19 +155,38 @@ async function getServiceAESKey(name, encoding) {
 }
 
 async function setServicePublicKey(name, value) {
-	await setCachedFile('service', name, 'public_key', value);
+	await setCachedFile(`service:${name}:public_key`, value);
 }
 
 async function setServicePrivateKey(name, value) {
-	await setCachedFile('service', name, 'private_key', value);
+	await setCachedFile(`service:${name}:private_key`, value);
 }
 
 async function setServiceSecretKey(name, value) {
-	await setCachedFile('service', name, 'secret_key', value);
+	await setCachedFile(`service:${name}:secret_key`, value);
 }
 
 async function setServiceAESKey(name, value) {
-	await setCachedFile('service', name, 'aes_key', value);
+	await setCachedFile(`service:${name}:aes_key`, value);
+}
+
+// Local CDN cache functions
+
+async function getLocalCDNFile(name, encoding) {
+	let file = await getCachedFile(`local_cdn:${name}`, encoding);
+
+	if (file === null) {
+		if (await fs.pathExists(`${LOCAL_CDN_BASE}/${name}`)) {
+			file = await fs.readFile(`${LOCAL_CDN_BASE}/${name}`, { encoding });
+			await setLocalCDNFile(name, file);
```

Comment:
How big are the CDN files (mii images, right?) We should make it clear that this is a dev-only feature to stop people running servers and running out of memory.

Have you checked if this is actually faster for large files? Small stuff sure, but OSes are already pretty good at disk caching these days. I'm not sure we should even bother with caching for a dev thing.

Reply:
Right now the server can technically support any sized file since the `datastore` service makes use of these functions too now. But that service will be removed in the future leaving, yes, only Mii images for now. I did some light testing with different sized files and reading from memory was always faster, but I'm aware this could be a classice case of "works on my system"

As for making it clear it's a dev-only feature, when the memory cache is being used the console prints out a warning saying it's using an in-memory cache. Any ideas on how to make that more clear?

Reply:
I am specifically thinking about the case with no S3, rather than no Redis (which still uses memory proportional to the filesizes, right? or does it offload things to disk under pressure?)
> s3 file storage disabled. Using disk-based file storage. Please ensure cdn_base config or PN_ACT_CONFIG_CDN_BASE env variable is set to point to this server with the subdomain being ${config.cdn_subdomain}

is what we say right now in that case.

Maybe something like
> Both s3 and Redis are disabled. Large CDN files will use the in-memory cache, which may result in high memory use. Please enable [s3 I assume] if you're running a production server.

Reply:
Oh my apolgies I misunderstood. Redis does not dump anything to disk by default last I checked, though it can be configured to do so

That warning message is very clear, will do